packages/mc-http-server/routes/login.js

```diff
@@ -1,7 +0,0 @@
-module.exports = (env) => (request, response, next) => {
-  if (!env.servedByProxy) {
```

Comment:
This was the original issue, where the `disableAuthRoutesOfDevServer` wasn't taken into account.

With the new config, the `servedByProxy` value is inferred from the node env or app env.
In the case of the auth app, when starting the app locally in production mode (`NODE_ENV=production MC_APP_ENV=development`), the `servedByProxy` is set to false, causing the dev login page to be rendered. However, for the auth app, we want to render the "real" login page and for that we need to use the `disableAuthRoutesOfDevServer`.